kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -102,8 +98,9 @@ public TopicPartitionWriter(TopicPartition tp,
 
     flushSize = connectorConfig.getInt(S3SinkConnectorConfig.FLUSH_SIZE_CONFIG);
     topicsDir = connectorConfig.getString(StorageCommonConfig.TOPICS_DIR_CONFIG);
-    rotateIntervalMs = connectorConfig.getLong(S3SinkConnectorConfig.ROTATE_INTERVAL_MS_CONFIG);
-    rotateScheduleIntervalMs = connectorConfig.getLong(S3SinkConnectorConfig.ROTATE_SCHEDULE_INTERVAL_MS_CONFIG);
+    rotateIntervalMs = partitioner instanceof TimeBasedPartitioner ?
```

Comment:
Yeah, I guess rotate interval can't really be used if you want exactly once. Some folks may be ok without that guarantee, especially if they can guarantee the latency of delivery.